cachito/workers/pkg_managers/pip.py

```diff
@@ -1575,3 +1578,104 @@ def get_hosted_repositories_username(request_id):
+        this option will be omitted
+    :return: the contents of the pip.conf file
+    :rtype: str
+    """
+    # Instead of getting the token from Nexus, use basic authentication as supported by Pip:
+    # https://pip.pypa.io/en/stable/user_guide/#basic-authentication-credentials
+    # Create the request specific nexus repo url to be passed as `index-url` within pip.conf
+    if "://" in nexus_pypi_hosted_repo_url:
+        nexus_pypi_hosted_repo_url = nexus_pypi_hosted_repo_url.replace(
+            "://", f"://{username}:{password}@", 1
+        )
+    else:
+        raise CachitoError(
+            f"Nexus PyPI Hosted Repo URL: {nexus_pypi_hosted_repo_url} is not a valid URL"
+        )
+
+    pip_conf_content = textwrap.dedent(
+        f"""\
+        [global]
+        index-url = {nexus_pypi_hosted_repo_url}
+        """
+    )
+    if custom_ca_path:
+        # The CA could be embedded in the actual pip.conf file with the `cert` option, but this
+        # would mean that the CA file contents would be duplicated in the Cachito database for
+        # every request that uses a pip conf file
+        pip_conf_content += f'cert = "{custom_ca_path}"\n'
+
+    return pip_conf_content
+
+
+def create_custom_pip_conf_file(request_id, package_configs=None):
+    """
+    Create a custom pip.conf file.
+
+    It should configure pip to point to the request specific repository in Nexus.
+
+    :param int request_id: the ID of the Cachito request
+    :param list package_configs: the list of optional package configurations submitted by
+    the user
+    :rtype: list<dict>
+    :return: The list of files for the pip configuration files that point to the request
+    specific repository in Nexus
+    """
+    if package_configs is None:
+        package_configs = []
+
+    config = get_worker_config()
+    nexus_pypi_hosted_repo_url = get_pypi_hosted_repo_url(request_id)
+    ca_cert = nexus.get_ca_cert()
+    subpaths = [os.path.normpath(c["path"]) for c in package_configs if c.get("path")]
+    if not subpaths:
+        # Default to the root of the application source
+        subpaths = [os.curdir]
+
+    log.info("Generating the pip conf files")
+    pip_config_files = []
+    if ca_cert:
+        # The custom CA will be called ca.cert in the "app" directory
+        pip_config_files.append(
+            {
+                "content": base64.b64encode(ca_cert.encode("utf-8")).decode("utf-8"),
+                "path": os.path.join("app", "ca.cert"),
+                "type": "base64",
+            }
+        )
+
+    for subpath in subpaths:
+        if ca_cert:
+            # Determine the relative path to the ca.cert file
+            custom_ca_path = os.path.relpath("ca.cert", start=subpath)
+        else:
+            custom_ca_path = None
+        pip_conf_content = generate_pip_conf_content(
+            nexus_pypi_hosted_repo_url,
+            config.cachito_nexus_proxy_username,
+            config.cachito_nexus_proxy_password,
```

Comment:
I'm not completely sure about this, but maybe these should be
- username: what you pass to `finalize_nexus_for_pip_request()`
- password: what you get back from `finalize_nexus_for_pip_request()`

https://github.com/release-engineering/cachito/blob/a25aca2642767192ed1c2df3d86cd0118d4adbd8/cachito/workers/pkg_managers/pip.py#L1208

@athos-ribeiro ^ is this correct?

If it is correct, it would probably be better to accept username and password as params for this function.

Reply:
+1

The username and password here should be the temporary ones, for the request volatile repositories.

The fetch_sources function will need the auth sdata and will fetch it already. Receiving the params here should be enough, as Adam suggested.